internal/services/kusto/kusto_attached_database_configuration_resource.go

```diff
@@ -70,10 +71,11 @@ func resourceKustoAttachedDatabaseConfiguration() *pluginsdk.Resource {
 			},
 
 			"cluster_resource_id": {
-				Type:         pluginsdk.TypeString,
-				Required:     true,
-				ForceNew:     true,
-				ValidateFunc: azure.ValidateResourceID,
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateFunc:     azure.ValidateResourceID,
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
I have changed back to care casing difference. But this would cause following error in acc test. The root cause is service returned a id in lower case.

------- Stdout: -------
=== RUN   TestAccKustoAttachedDatabaseConfiguration_basic
=== PAUSE TestAccKustoAttachedDatabaseConfiguration_basic
=== CONT  TestAccKustoAttachedDatabaseConfiguration_basic
testcase.go:110: Step 1/2 error: After applying this test step, the plan was not empty.
stdout:
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement
Terraform will perform the following actions:
# azurerm_kusto_attached_database_configuration.test must be replaced
-/+ resource "azurerm_kusto_attached_database_configuration" "test" {
~ attached_database_names             = [
- "acctestkd2-221221151428955994",
] -> (known after apply)
~ cluster_resource_id                 = "/subscriptions/*******/resourceGroups/acctestRG-221221151428955994/providers/Microsoft.Kusto/Clusters/acctestkc2rgoe7" -> "/subscriptions/*******/resourceGroups/acctestRG-221221151428955994/providers/Microsoft.Kusto/clusters/acctestkc2rgoe7" # forces replacement
~ id                                  = "/subscriptions/*******/resourceGroups/acctestRG-221221151428955994/providers/Microsoft.Kusto/clusters/acctestkc1rgoe7/attachedDatabaseConfigurations/acctestka-221221151428955994" -> (known after apply)
name                                = "acctestka-221221151428955994"
# (5 unchanged attributes hidden)
# (1 unchanged block hidden)
}
Plan: 1 to add, 0 to change, 1 to destroy.
--- FAIL: TestAccKustoAttachedDatabaseConfiguration_basic (1749.32s)
FAIL